### PR TITLE
Minor source editor fixes

### PR DESCRIPTION
**Pull request by corani**
- Return to the same file after saving
- Reset caret to line 1 when changing file
- Consistent order of source files

### CODE REVIEW

File: fava/templates/source.html [hunk 1]
```diff
@@ -15,7 +15,7 @@
                 </tr>
             </thead>
             <tbody>
-                {% for source_file in api.source_files() %}
+                {% for source_file in api.source_files()|sort %}
```
Comment (aumayr) [inline fava/templates/source.html]: @corani `api.source_files()` returns the source files in the order that beancount loads them - so your main file first. Are you sure you want them to be sorted by name instead?

Comment (corani) [inline fava/templates/source.html]: What I've been noticing is that the order in which they're displayed (for me) is pretty random between restarts. Definitely not the order in which they're specified in the main source file. (I've got my files split by year and load them in order, yet in the UI they're random)

Comment (aumayr) [inline fava/templates/source.html]: Ah OK I see. Should be sorted here, then:

https://github.com/aumayr/fava/blob/master/fava/api/__init__.py#L455

Comment (corani) [inline fava/templates/source.html]: I think it's more of a representation thing, not really worth putting it in the API. I'm happy to change it though, if you still think it's better.

Comment (aumayr) [inline fava/templates/source.html]: The API should be consistent in what it returns, nonetheless, so I think it's worth changing.

File: fava/api/__init__.py [hunk 2]
```diff
@@ -451,9 +451,9 @@ def linechart_data(self, account_name):
         } for journal_entry in journal if 'balance' in journal_entry.keys()]
 
     def source_files(self):
-        return list(set(
```
Comment (corani) [inline fava/api/__init__.py]: A `set` is unordered, using it to remove the duplicate main source file causes the ordering to get lost.

**Comment by aumayr** [comment]
Perfect, thanks!

**Pull request merged**